pages/videos.tsx

```diff
@@ -0,0 +1,111 @@
+import PopupVideo from '@components/Video/PopupVideo'
+import VideoCard from '@components/Video/VideoCard'
+import { Fragment, useState } from 'react'
+import { Dialog, Transition } from '@headlessui/react'
+
+interface Video {
+  youtubeVideoID: string
+  title: string
+}
+
+interface VideosGridProps {
+  videos: Array<Video>
+}
+
+const VideosGridView: React.FC<VideosGridProps> = ({ videos }) => {
+  videos = [
+    {
+      youtubeVideoID: 's9F8pu5KfyM',
+      title: 'Why Do Computers Suck At Math?'
+    },
+    {
+      youtubeVideoID: 'B1t4Fjlomi8',
+      title: 'Why do computers use RGB for colours, and not RBY?'
+    },
+    {
+      youtubeVideoID: 'njdJeu95p6s',
+      title: 'Top 3 Ways to Center a DIV with CSS'
+    },
+    {
+      youtubeVideoID: 'njdJeu95p6s',
+      title: 'Top 3 Ways to Center a DIV with CSS'
+    }
+  ]
+
+  const [popupVisible, setPopupVisibility] = useState(false)
+  const handlePopupVisibility = () => setPopupVisibility(!popupVisible)
+  const [activeVideo, setActiveVideo] = useState(
+    'https://www.youtube.com/watch?v=oUVCWNQFGTc'
+  )
+  const handleActiveVideoChange = (url: string) => {
+    setActiveVideo(url)
+    setPopupVisibility(true)
+  }
+
+  const videoCards = videos.map(video => {
+    const { youtubeVideoID, title } = video
+    return (
+      <button
+        className='bg-primary flex flex-col rounded-2xl overflow-hidden'
+        key={youtubeVideoID}
+        onClick={() =>
+          handleActiveVideoChange(
+            'https://www.youtube.com/watch?v=' + youtubeVideoID
+          )
+        }
+      >
+        <VideoCard title={title} youtubeVideoID={youtubeVideoID} />
+        <h1 className=' px-4 py-1 font-serif text-md text-white hover:opacity-75'>
+          {title}
+        </h1>
+      </button>
```

Comment:
As a general rule its best to create a React component when you iterate over an array to dynamically create elements to make it easier to read. e.g. `<VideoCardContainer video={video} onClick={handleVideoClick} />`